src/pe.rs

```diff
@@ -0,0 +1,2984 @@
+pub const IMAGE_ARCHIVE_LONGNAMES_MEMBER: &[u8; 16] = b"//              ";
+pub const IMAGE_ARCHIVE_HYBRIDMAP_MEMBER: &[u8; 16] = b"/<HYBRIDMAP>/   ";
+
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct ImageArchiveMemberHeader {
+    /// File member name - `/' terminated.
+    pub name: [u8; 16],
+    /// File member date - decimal.
+    pub date: [u8; 12],
+    /// File member user id - decimal.
+    pub user_id: [u8; 6],
+    /// File member group id - decimal.
+    pub group_id: [u8; 6],
+    /// File member mode - octal.
+    pub mode: [u8; 8],
+    /// File member size - decimal.
+    pub size: [u8; 10],
+    /// String to end header.
+    pub end_header: [u8; 2],
+}
+
+pub const IMAGE_SIZEOF_ARCHIVE_MEMBER_HDR: u16 = 60;
+
+//
+// DLL support.
+//
+
+//
+// Export Format
+//
+
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct ImageExportDirectory {
+    pub characteristics: U32<LE>,
+    pub time_date_stamp: U32<LE>,
+    pub major_version: U16<LE>,
+    pub minor_version: U16<LE>,
+    pub name: U32<LE>,
+    pub base: U32<LE>,
+    pub number_of_functions: U32<LE>,
+    pub number_of_names: U32<LE>,
+    /// RVA from base of image
+    pub address_of_functions: U32<LE>,
+    /// RVA from base of image
+    pub address_of_names: U32<LE>,
+    /// RVA from base of image
+    pub address_of_name_ordinals: U32<LE>,
+}
+
+//
+// Import Format
+//
+
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct ImageImportByName {
+    pub hint: U16<LE>,
+    //pub name: [i8; 1],
+}
+
+/*
+
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct ImageThunkData64 {
+    union {
+/// PBYTE
+        pub forwarder_string: U64<LE>,
+/// PDWORD
+        pub function: U64<LE>,
+        pub ordinal: U64<LE>,
+/// PIMAGE_IMPORT_BY_NAME
+        pub address_of_data: U64<LE>,
+    } u1;
+}
+
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct ImageThunkData32 {
+    union {
+/// PBYTE
+        pub forwarder_string: U32<LE>,
+/// PDWORD
+        pub function: U32<LE>,
+        pub ordinal: U32<LE>,
+/// PIMAGE_IMPORT_BY_NAME
+        pub address_of_data: U32<LE>,
+    } u1;
+}
+
+pub const IMAGE_ORDINAL_FLAG64: u64 = 0x8000000000000000;
+pub const IMAGE_ORDINAL_FLAG32: u32 = 0x80000000;
+#define IMAGE_ORDINAL64(Ordinal) (Ordinal & 0xffff)
+#define IMAGE_ORDINAL32(Ordinal) (Ordinal & 0xffff)
+#define IMAGE_SNAP_BY_ORDINAL64(Ordinal) ((Ordinal & IMAGE_ORDINAL_FLAG64) != 0)
+#define IMAGE_SNAP_BY_ORDINAL32(Ordinal) ((Ordinal & IMAGE_ORDINAL_FLAG32) != 0)
+
+*/
+
+//
+// Thread Local Storage
+//
+
+/*
+typedef VOID
+(NTAPI *PIMAGE_TLS_CALLBACK) (
+    PVOID DllHandle,
+    DWORD Reason,
+    PVOID Reserved
+    );
+    */
+
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct ImageTlsDirectory64 {
+    pub start_address_of_raw_data: U64<LE>,
+    pub end_address_of_raw_data: U64<LE>,
+    /// PDWORD
+    pub address_of_index: U64<LE>,
+    /// PIMAGE_TLS_CALLBACK *;
+    pub address_of_call_backs: U64<LE>,
+    pub size_of_zero_fill: U32<LE>,
+    
```

Comment:
Why is this commented out?

Reply:
Because I don't need it yet, and it has bitfields so it's not trivial to do.